roofit/roofitcore/inc/RooFuncWrapper.h

```diff
@@ -84,21 +88,38 @@ class RooFuncWrapper final : public RooAbsReal {
       }
       _gradientVarBuffer.reserve(_params.size());
 
+      // Extract observables
+      if (!ObsSet.empty()) {
+         auto dataSpans = data.getBatches(0, data.numEntries());
+
+         for (auto *obs : static_range_cast<RooRealVar *>(ObsSet)) {
+            RooSpan<const double> span{dataSpans.at(obs)};
+            for (int i = 0; i < data.numEntries(); ++i) {
+               _observables.push_back(span[i]);
+            }
+         }
+      }
+
       // Build a wrapper over the derivative to hide clad specific types such as 'array_ref'.
       // disable clang-format for making the following code unreadable.
       // clang-format off
       std::stringstream dWrapperStrm;
-      dWrapperStrm << "void " << wrapperName << "(double* in, double* out) {\n"
+      dWrapperStrm << "void " << wrapperName << "(double* in, double* obs, double* out) {\n"
```

Comment:
```suggestion
      dWrapperStrm << "void " << wrapperName << "(double* params, double* obs, double* out) {\n"
```
To be more consistent with the other names?